js/flightlog.js

```diff
@@ -450,8 +455,38 @@ function FlightLog(logData) {
                                 // But other data from the G frame can be valid (time, num sats)
+
+                                //H Field G name:time,GPS_numSat,GPS_coord[0],GPS_coord[1],GPS_altitude,GPS_speed,GPS_ground_course
+                                for (let i = 0; i < frame.length; i++) {
+                                    lastGPS[i] = frame[i];
+                                }
+                                console.log("frameValid: " + frameValid);
+                                console.log("frame: " + frame);
+                                console.log("frameType: " + frameType);
+                                console.log("frameOffset: " + frameOffset);
+                                console.log("frameSize: " + frameSize);
+
+                                    //parser.onFrameReady = function(frameValid, frame, frameType, frameOffset, frameSize) {
+//                                console.log("local frames", chunk.frames[chunk.frames.length - 1]);
+//                                chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[0]')] = lastGPS[0];
+//                                chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[1]')] = lastGPS[1];
+//                                chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[2]')] = lastGPS[2];
+//                                chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[3]')] = lastGPS[3];
+//                                chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[4]')] = lastGPS[4];
+//                                chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[5]')] = lastGPS[5];
+                                //chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[6]')] = lastGPS[6];
+
+                                //getMainFieldIndexByName
                             break;
                         }
+                            // Logs gps values, but out of sync for some reason
+                            console.log("local frames", chunk.frames[chunk.frames.length - 1]);
+                            chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[0]')] = lastGPS[0];
+                            chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[1]')] = lastGPS[1];
+                            chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[2]')] = lastGPS[2];
+                            chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[3]')] = lastGPS[3];
+                            chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[4]')] = lastGPS[4];
+                            chunk.frames[chunk.frames.length - 1][that.getMainFieldIndexByName('lastGPS[5]')] = lastGPS[5];
```

Comment:
I think I understand what you are doing here, I suppose you want to "store" the latest known GPS data in the standard chunk. If I'm right, I think that what you want is to do the same that it does for the S (Slow) frame upper, in the I and P frames, look for this comment `// Then merge in the last seen slow-frame data` and do the same to add the GPS data to the standard chunk.

Reply:
Done.